code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardUIProvider.kt

```diff
@@ -0,0 +1,163 @@
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * ContentCardUiProvider is responsible for fetching and managing the content for a given surface.
+ * It uses Adobe Messaging APIs to retrieve propositions and transform them into UI templates for display.
+ *
+ * @property surfaceString The identifier of the surface for which content needs to be fetched.
+ */
+class ContentCardUIProvider(val surfaceString: String) : AepUIContentProvider {
+    companion object {
+        private const val SELF_TAG: String = "ContentCardUIProvider"
+    }
+
+    private val _contentFlow = MutableStateFlow<List<AepUITemplate>>(emptyList())
+    private val contentFlow: StateFlow<List<AepUITemplate>> = _contentFlow
+
+    private val _aepUiFlow = MutableStateFlow<List<AepUI<*, *>>>(emptyList())
+    private val aepUiFlow: StateFlow<List<AepUI<*, *>>> = _aepUiFlow
+
+    /**
+     * Retrieves a flow of AepUI instances for the given surface.
+     *
+     * This function initiates the content fetch using [getContent] and then returns a flow of
+     * [AepUI] instances that represent the UI templates. The flow emits updates whenever new
+     * content is fetched or any changes occur.
+     *
+     * @return A [Flow] that emits a list of [AepUI] instances.
+     */
+    suspend fun getAepUi(): Flow<List<AepUI<*, *>>> {
+        getContent()
+        return aepUiFlow
+    }
+
+    /**
+     * Clears the current content and re-fetches new content for the given surface.
+     */
+    override suspend fun refreshContent() {
+        _contentFlow.value = emptyList()
+        _aepUiFlow.value = emptyList()
+        getContent()
+    }
+
+    /**
+     * Retrieves a flow of AepUITemplate lists for the given surface.
+     * The flow emits updates whenever new content is fetched.
+     *
+     * @return A flow that emits lists of AepUITemplate.
+     */
+    override suspend fun getContent(): Flow<List<AepUITemplate>> {
+        CoroutineScope(Dispatchers.IO).launch {
+            val surface = Surface(surfaceString)
+            val surfaceList = mutableListOf<Surface>()
+            surfaceList.add(surface)
+            getAepUITemplateList(surfaceList) { it ->
+                it.onSuccess { templateList ->
+                    _contentFlow.value = templateList
+                    _aepUiFlow.value = templateList.mapNotNull { item -> getAepUI(item) }
+                }
+                it.onFailure { error ->
+                    Log.error(
+                        MessagingConstants.LOG_TAG, SELF_TAG,
+                        "Failed to get content: ${error.message}"
+                    )
+                    _contentFlow.value = emptyList()
+                    _aepUiFlow.value = emptyList()
+                }
+            }
+        }
+        return contentFlow
+    }
+
+    private suspend fun getAepUITemplateList(
+        surfaceList: MutableList<Surface>,
```

Comment:
Still not sure why this needs to be a list. Please refer to this comment
https://github.com/adobe/aepsdk-messaging-android/pull/292#discussion_r1819689273